build.py

```diff
@@ -113,20 +113,71 @@ def get_distribution_name():
     except ImportError:
         return distro.codename()
 
-def get_package_filename(p):
+def get_distribution_type(distro_id = None):
+    if distro_id is None:
+        distro_id = distro.id()
+    if distro_id == 'ol': # special handling for Oracle Linux
+        return 'rhel'
+    if distro_id in ['rhel', 'ubuntu', 'suse']:
+        return distro_id
+    distro_like = distro.like().split()
+    if len(distro_like) == 0:
+        return distro_id
+    return get_distribution_type(distro_like[0])
+
+# Does not account for derivatives that follow their own version scheme
+# (ex: RHEL7-like Amazon Linux 2)
+def get_distribution_version():
+    distro_type = get_distribution_type()
+    if distro_type == 'ubuntu':
+        return '{0}.{1}'.format(distro.major_version(), distro.minor_version())
+    elif distro_type == 'suse':
+        return distro.version()
+    distro_version = distro.major_version()
+    if distro_version in ['', None]:
+        return 'rolling'
```

Comment:
Can you explain the use of `rolling` here?

Reply:
Typically, if there is no distribution version, it's a rolling release distro, like arch.

Reply:
Cool cool.